File: Java - Backend - Javalin/src/main/java/backend/Server_router.java
package backend;

import io.javalin.Javalin;
import java.util.List;
import java.util.Map;

import org.mindrot.jbcrypt.BCrypt;

import org.jooq.DSLContext;
import org.jooq.impl.DSL;
import org.jooq.SelectJoinStep;
import org.jooq.Result;

import database.*;



public class Server_router {
    public static Javalin exe() {

        Javalin server = Server.create();
        

        // ========================================= //
        // ================ Welcome ================ //
        // ========================================= //
        server.get("/", ctx -> {
            ctx.result("Welcome"); 
        });
        
        
        // ======================================== //
        // ================ Parser ================ //
        // ======================================== //
        server.get("/parser/{slug1}/{slug2}", ctx -> {
            
            String slug1  = ctx.pathParam("slug1");
            String slug2  = ctx.pathParam("slug2");
            String param1 = ctx.queryParam("param1");
            String param2 = ctx.queryParam("param2");

            ctx.json(Map.of(
                "slug1",  slug1,
                "slug2",  slug2,
                "param1", param1,
                "param2", param2
            ));
        });


        // ====================================== //
        // ================ Auth ================ //
        // ====================================== //
        server.get("/auth-pwd-code", ctx -> {  
            String password = "ABC123"; 
            String hashed   =  BCrypt.hashpw(password, BCrypt.gensalt());
            ctx.json(Map.of("hashed", hashed));
        }); 
        
        
        server.get("/auth-pwd-check", ctx -> {  
            String  password = "ABC123";
            String  hashed   = "$2a$10$NEthZtIcabO02lR65OcJz.qsvIXee0YCdtl5pJ86vSbqQC3sQBVje";
            Boolean isValid  =  BCrypt.checkpw(password, hashed); 
            ctx.json(Map.of("isValid", isValid));
        }); 

        
        server.get("/auth-jwt-code", ctx -> { 
            String email = "user@mail";
            String token = Auth.jwt_code(Map.of("email", email), 7*(24*60*60));
            ctx.json(Map.of("token", token));
        });
        
        
        server.get("/auth-jwt-check", ctx -> {  
            String  token   = "eyJhbGciOiJIUzUxMiJ9.eyJlbWFpbCI6InVzZXJAbWFpbCIsImlhdCI6MTc0MTE0ODMwOCwiZXhwIjoxNzQxNzUzMTA4fQ.CZ3NK_YCRBLA68jFaflWRqb4TFH1WoiZMeF3Ug4cZ2UEFx6rcZLvG61esxrOD43YOUpC6y1GvZIw2zOtBpQyxg";
            Boolean isValid = Auth.jwt_check(token);
            ctx.json(Map.of("isValid", isValid));
        });


        // ========================================== //
        // ================ Database ================ //
        // ========================================== //
        server.get("/dbase-read", ctx -> {  
            try {
                DSLContext      db_con  = _Connection.exe();
                SelectJoinStep  query   = db_con.select().from(DSL.table("'Table'"));
                List            res     = query.fetch().intoMaps();
                ctx.json(res) ;
            } catch (Exception E) {
                ctx.result("Exception"); 
            }
        });
        
        server.get("/dbase-bind", ctx -> {
            try {
                String     SQL    = "SELECT * FROM 'Table' WHERE ColA = ?";
                String     search = ctx.queryParam("search");
                DSLContext db_con = _Connection.exe();
                Result     query  = db_con.fetch(SQL, search);
                List       res    = query.intoMaps();
                ctx.json(res);
            } catch (Exception E) {
                ctx.result("Exception");
            }
        });

        
        return server;
    }
}

File: Kotlin - Backend - Javalin/app/src/main/kotlin/backend/Server_router.kt
package backend

import io.javalin.Javalin
import org.jooq.impl.DSL

import kotlinx.serialization.*
import kotlinx.serialization.json.*

import backend.*
import database.*



fun Server_router(): Javalin {
    val server = Server()


    // ========================================== //
    // ================ Wellcome ================ //
    // ========================================== //
    server.get("/") { ctx -> 
        ctx.result("Welcome") 
    }
    

    // ========================================= //
    // ================ Req/Res ================ //
    // ========================================= //
    server.get("/parser/{slug1}/{slug2}") { ctx ->
        
        val slug1  = ctx.pathParam("slug1")
        val slug2  = ctx.pathParam("slug2")
        val param1 = ctx.queryParam("param1")
        val param2 = ctx.queryParam("param2")

        ctx.json(mapOf(
            "slug1"  to slug1,
            "slug2"  to slug2,
            "param1" to param1,
            "param2" to param2
        ))
    }
    

    // ====================================== //
    // ================ Auth ================ //
    // ====================================== //
    server.get("/auth-pwd-code") { ctx -> 
        val password = "ABC123"
        val hashed   = Auth_pwd_code(password)
        ctx.json(mapOf("hashed" to hashed))
    }
    
    
    server.get("/auth-pwd-check") { ctx -> 
        val password = "ABC123"
        val hashed   = "SOME_HASH"
        val result   = Auth_pwd_check(password, hashed) 
        ctx.json(result)
    }

    
    server.get("/auth-jwt-code") { ctx -> 
        val content = mapOf("email" to "user@mail")
        val token = Auth_jwt_code(Json.encodeToString(content))
        ctx.json(mapOf("token" to token))
    }
    
    
    server.get("/auth-jwt-check") { ctx -> 
        val token  = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJJc3N1ZXIiLCJzdWIiOiJTdWJqZWN0IiwiQ2xhaW0iOiJ7XCJlbWFpbFwiOlwidXNlckBtYWlsXCJ9IiwiZXhwIjoxNzQxNzQzNzUwfQ.o0sB1d8gSbJrgolhbTsmbl0HV0iiRJS_BmLqmOXW0x4"
        val result = Auth_jwt_check(token)
        ctx.json(result)
    }


    // ========================================== //
    // ================ DATABASE ================ //
    // ========================================== //
    server.get("/dbase-read") { ctx -> 
        val db_con  = Connection()
        val query   = db_con.select().from(DSL.table("'Table'"))
        val res     = query.fetch().intoMaps()      
        ctx.json(res) 
    }
    
    server.get("/dbase-bind") { ctx ->
        val SQL     = """
            SELECT * 
            FROM 'Table' 
            WHERE ColA = ?
        """
        val search  = ctx.queryParam("search")
        val db_con  = Connection()
        val query   = db_con.fetch(SQL, search)
        val res     = query.intoMaps()
        ctx.json(res) 
    }


    return server
}